cider/tests/JITLibTest.cpp

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include "exec/nextgen/jitlib/JITLib.h"
+
+#include <gtest/gtest.h>
+
+#include "tests/TestHelpers.h"
+
+using namespace jitlib;
+
+class JITLibTests : public ::testing::Test {};
+
+TEST_F(JITLibTests, BasicTest) {
```

Comment:
Providing another two examples will be very helpful.
(1) nested modules (if valid)
(2) nested functions within the same functions.

Reply:
There is no nested modules, but I can add some cases about nested functions.